sidechain/consensus/common/src/block_import_queue_worker.rs

```diff
@@ -96,25 +100,43 @@ impl<ParentchainBlock, SignedSidechainBlock, BlockImportQueue, PeerBlockSyncer>
 		let mut number_of_imported_blocks = 0usize;
 		let start_time = Instant::now();
 
-		loop {
-			match self.block_import_queue.pop_front() {
-				Ok(maybe_block) => match maybe_block {
-					Some(block) => {
-						latest_imported_parentchain_header = self
-							.peer_block_syncer
-							.sync_block(block, &latest_imported_parentchain_header)?;
-						number_of_imported_blocks += 1;
-					},
-					None => {
-						Self::record_timings(start_time, number_of_imported_blocks);
-						return Ok(latest_imported_parentchain_header)
-					},
-				},
-				Err(e) => {
-					Self::record_timings(start_time, number_of_imported_blocks);
-					return Err(Error::FailedToPopBlockImportQueue(e))
-				},
-			}
+		trace!(
+			"processing import queue with {:?} sidechain blocks",
+			self.block_import_queue.peek_queue_size()
+		);
+
+		if let Ok(candidates) = self.block_import_queue.pop_all() {
```

Comment:
is it ok to not hanlde the error case? it would also reduce the code nesting if the could just do:

```rust
self.block_import_queue.pop_all()?
   .iter()...
```

Reply:
pretty sure you're right. let's see what CI tells us